test/visitor/local_to_assigned.cpp

```diff
@@ -0,0 +1,80 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include "catch/catch.hpp"
+
+#include "ast/program.hpp"
+#include "parser/nmodl_driver.hpp"
+#include "test/utils/nmodl_constructs.hpp"
+#include "visitors/local_var_visitor.hpp"
+#include "visitors/lookup_visitor.hpp"
+#include "visitors/nmodl_visitor.hpp"
+#include "visitors/perf_visitor.hpp"
+#include "visitors/symtab_visitor.hpp"
+
+using namespace nmodl;
+using namespace visitor;
+using namespace test_utils;
+
+using ast::AstNodeType;
+using nmodl::parser::NmodlDriver;
+using symtab::syminfo::NmodlType;
+
+//=============================================================================
+// GlobalToRange visitor tests
+//=============================================================================
+
+std::shared_ptr<ast::Program> run_local_to_assigned_visitor(const std::string& text) {
+    std::map<std::string, std::string> rval;
+    NmodlDriver driver;
+    auto ast = driver.parse_string(text);
+
+    SymtabVisitor().visit_program(*ast);
+    PerfVisitor().visit_program(*ast);
+    LocalToAssignedVisitor().visit_program(*ast);
+    SymtabVisitor().visit_program(*ast);
+    return ast;
+}
+
+SCENARIO("LOCAL to ASSIGNED variable transformer", "[visitor][localtoassigned]") {
+    GIVEN("mod file with LOCAL variables that are written") {
+        std::string input_nmodl = R"(
+            NEURON {
+                SUFFIX test
+            }
+            LOCAL x, y, z
```

Comment:
question: is it legal in NMODL to have a variable with same name in LOCAL and in the NEURON block?

Reply:
It's possible yes, however the variable with this visitor should be defined as `RANGE` otherwise the `CodegenCompatibilityVisitor` will complain with:
```
[NMODL] [error] :: Code Incompatibility :: "x" variable found at [3.8] should be defined as a RANGE variable instead of GLOBAL to enable backend transformations
```